src/DynamoCoreWpf/ViewModels/Search/BrowserItemViewModel.cs

```diff
@@ -444,10 +447,40 @@ private void AddToItems(IEnumerable<ISearchEntryViewModel> newItems)
                 if (entry is ClassesNodeCategoryViewModel)
                     Items.Insert(0, entry);
                 else
-                    if (first != null)
-                        Items.Insert(first.Idx, entry);
```

Comment:
Bulk of the following conditions are identical, here's how you can shorten it:

```csharp
if (entry is NodeSearchElementViewModel)
{
    if (first != null)
        Items.Insert(first.Idx + SubCategories.Count(), entry);
    else
        Items.Add(entry);
}
if (entry is NodeCategoryViewModel)
{
    if (first != null)
    {
        var offset = hasClasses ? 1 : 0;
        Items.Insert(first.Idx + offset, entry);
    }
    else
        Items.Insert(Items.Count - Entries.Count, entry);
}
```

Also, `SubCategories.Count()` forces `IEnumerable` to enumerate for results, perhaps it's a better idea to move `SubCategories.Count()` out of the loop, if it does not change for the duration of the loop.